### PR TITLE
[inductor] Update triton pin

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114772

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D51761353](https://our.internmc.facebook.com/intern/diff/D51761353)

### CODE REVIEW

**Comment by davidberard98** [comment]
cc @peterbell10

**Review by shunting314 — APPROVED** [review]
Has the triton issues that block our previous pin move being resolved?

Also I'm curious if we see some perf movement by moving the pin since the current pin is a few months back.

**Comment by desertfire** [comment]
> Has the triton issues that block our previous pin move being resolved?
> 
> Also I'm curious if we see some perf movement by moving the pin since the current pin is a few months back.

According to @davidberard98 , https://github.com/openai/triton/pull/2608 (the triton commit id used here) fixes a previous blocker. I will test the functionality first and then kick off a dashboard perf run.

**Comment by lezcano** [comment]
FWIW, Peter is on PTO this week, he'll be back on Monday.
There are still some issues when updatin the pin (as you can see in CI). In particular, there is one that was particularly tricky to narrow down. Peter has been looking into finding a self-contained repro, but it's proving tricky, as it's a non-deterministic one, so the minimiser doesn't quite work. If any of you guys wants to help, feel free to reach out to him on PyTorch's slack or here

**Comment by atalman** [comment]
Failure: [linux-binary-manywheel / manywheel-py3_8-cuda12_1-test / test](https://hud.pytorch.org/pr/pytorch/pytorch/114772#19222801218) ([gh](https://github.com/pytorch/pytorch/actions/runs/7060297374/job/19222801218))
can be ignored, its chicken and egg problem, its now looking for new triton package which is not there yet

**Comment by desertfire** [comment]
The [dashboard run ](https://hud.pytorch.org/benchmark/compilers?startTime=Fri%2C%2024%20Nov%202023%2015%3A57%3A04%20GMT&stopTime=Fri%2C%2001%20Dec%202023%2015%3A57%3A04%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/desertfire/278/head&lCommit=94bf6e9b42418c738686f3d4f536c1680de46587&rBranch=main&rCommit=7f4064034266d860a79fa7dbeaba30fde159b038) looks good. There is no correctness or performance regression. There is some perf improvement for TIMM.

The `gmixer_24_224` accuracy failure in https://github.com/pytorch/pytorch/actions/runs/7060297733/job/19222950439 could be a slight accuracy difference. I can't reproduce locally and the dashboard run also shows ok. We can add `gmixer_24_224` to the `REQUIRE_HIGHER_TOLERANCE` set.

**Comment by davidberard98** [comment]
@desertfire do you know if this change will break inductor on old triton builds (e.g. internal) when it lands (due to the changes in inductor)?

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51761353).

**Review by atalman — APPROVED** [review]

**Comment by malfet** [comment]
Hmm, it broke ROCm builds :(

**Comment by malfet** [comment]
Also, after this change `rocm` triton packages will now upload as just `triton`, let me figure out how to fix it
